### PR TITLE
IoT MQTT Tests

### PR DESCRIPTION
**Pull request by gguuss**
* Adds setConfiguration to Manager for sending device configurations
* Adds tests for MQTT
* Adds subscribe to configuration updates for MQTT
* Prefer String.format over concatenation
* Update RSA certs to not expire (private keys are torn down after tests run)
* Consolidates device and manager projects as well as READMEs

### CODE REVIEW

File: iot/api-client/manager/README.md [hunk 1]
```diff
@@ -211,3 +211,56 @@ To publish state messages, run the sample as follows:
     gcloud beta pubsub subscriptions pull --auto-ack \
         projects/my-iot-project/subscriptions/my-subscription
 ```
+
+# Cloud IoT Core Java MQTT example
```
Comment (lesv) [inline iot/api-client/manager/README.md]: Seems very long not to put an index at the top?

File: iot/api-client/manager/pom.xml [hunk 2]
```diff
@@ -36,7 +36,19 @@
...
+      <id>Eclipse Paho Repo</id>
+      <url>https://repo.eclipse.org/content/repositories/paho-releases/</url>
+    </repository>
+  </repositories>
+
   <dependencies>
+    <dependency>
+      <groupId>org.eclipse.paho</groupId>
+      <artifactId>org.eclipse.paho.client.mqttv3</artifactId>
+      <version>1.0.2</version>
```
Comment (lesv) [inline iot/api-client/manager/pom.xml]: This appears in maven central and is currently at 1.2.0

File: iot/api-client/manager/pom.xml [hunk 3]
```diff
@@ -36,7 +36,19 @@
     <maven.compiler.target>1.7</maven.compiler.target>
   </properties>
 
+  <repositories>
+    <repository>
+      <id>Eclipse Paho Repo</id>
+      <url>https://repo.eclipse.org/content/repositories/paho-releases/</url>
```
Comment (lesv) [inline iot/api-client/manager/pom.xml]: don't think this is required as it's in [Maven Central](http://search.maven.org/#search%7Cga%7C1%7Corg.eclipse.paho.client.mqttv3)

File: iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java [hunk 4]
```diff
@@ -144,8 +146,8 @@ public static void deleteRegistry(String cloudRegion, String projectId, String r
         GoogleNetHttpTransport.newTrustedTransport(),jsonFactory, init)
         .setApplicationName(APP_NAME).build();
 
-    String projectPath = "projects/" + projectId + "/locations/" + cloudRegion;
-    String registryPath = projectPath + "/registries/" + registryName;
+    final String registryPath = String.format("projects/%s/locations/%s/registries/%s",
```
Comment (lesv) [inline iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java]: YES!

File: iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java [hunk 5]
```diff
@@ -343,4 +371,119 @@ public void testHttpDeviceConfig() throws Exception {
...
+    String[] testArgs = {
+        "-project_id=" + PROJECT_ID,
+        "-registry_id=" + REGISTRY_ID,
+        "-device_id=" + deviceName,
+        "-private_key_file=" + PKCS_PATH,
+        "-message_type=events",
+        "-num_messages=1",
+        "-algorithm=RS256"
+    };
+    com.example.cloud.iot.examples.MqttExample.main(testArgs);
```
Comment (lesv) [inline iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java]: I'm surprised, but I like this approach.
@jabubake @kurtisvg PTAL

File: iot/api-client/manager/pom.xml (hunk 3 above)
Comment (gguuss) [reply iot/api-client/manager/pom.xml]: Done.

File: iot/api-client/manager/pom.xml (hunk 2 above)
Comment (gguuss) [reply iot/api-client/manager/pom.xml]: Done.

File: iot/api-client/manager/README.md (hunk 1 above)
Comment (gguuss) [reply iot/api-client/manager/README.md]: Acknowledged, I'll have to clean up a few other things in the sample folder soon, I'll split up the READMEs when I do this.

**Comment by gguuss** [comment]
AFAICT, Vision Beta is breaking the build, I'll try and fix.

**Comment by lesv** [comment]
Note - was all green before update

**Comment by gguuss** [comment]
Hrrm, if you wannt me to roll back the change to the flaky vision test, I can, however, I think this fixed it.

**Comment by lesv** [comment]
@gguuss Ok to merge.  If it's Vision, then I don't care ATM.  We'll deal w it in 2018.

**Pull request merged**